### PR TITLE
Offline unittests

### PR DESCRIPTION
**Pull request by pde**
Ensure that unit tests can be run even if you don't have Internet access.

### CODE REVIEW

**Comment by bmw** [comment]
I can confirm that unittests currently fail on `master` if you don't have internet access. I can also confirm that simply adding `host='127.0.0.1'` is all that is needed to make the tests work. No new `mock` objects are necessary.

File: acme/acme/challenges_test.py [hunk 1]
```diff
@@ -359,11 +359,14 @@ def test_simple_verify(self, mock_verify_cert):
...
-    def test_simple_verify_false_on_probe_error(self):
+    @mock.patch('acme.challenges.socket.gethostbyname')
+    def test_simple_verify_false_on_probe_error(self, mock_gethostbyname):
         chall = mock.Mock()
         chall.probe_cert.side_effect = errors.Error
+
+        mock_gethostbyname.return_value = '127.0.0.2'
         self.assertFalse(self.msg.simple_verify(
-            self.chall, self.domain, self.key.public_key()))
+            self.chall, self.domain, self.key.public_key(), host="127.0.0.1"))
```
Comment (kuba) [inline acme/acme/challenges_test.py]: `probe_cert` is the only function in the entire module that does `gethostbyname` and it should already be mocked in `chall.probe_cert.side_effect = errors.Error` line. If it doesn't work then the whole test suite seems to be broken. So you're not fixing the root cause.

Indeed, we call `self.msg.simple_verify(self.chall, ...)` instead `self.msg.simple_verify(chall, ...)`.

Comment (kuba) [inline acme/acme/challenges_test.py]: In summary the only change in this PR should be: `self.chall` to `chall`. `host=...` should also be removed.

**Comment by kuba** [comment]
So, this did decrease coverage. I'm happy to finish up this PR, @pde.

**Comment by kuba** [comment]
FTR: `acme/challenges.py               215      5    98%   442-446` which corresponds to https://github.com/letsencrypt/letsencrypt/blob/890a76ca41d3187601f96599774267c93db5a823/acme/acme/challenges.py#L441-L446.

**Comment by kuba** [comment]
@pde: fixed in #1187.

**Comment by kuba** [comment]
#1187 has been pulled in, so allow me to close this one.

**Pull request closed without merging**